internal/handlers/assisted_service_client.go

```diff
@@ -0,0 +1,143 @@
+package handlers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	log "github.com/sirupsen/logrus"
+)
+
+type AssistedServiceClient struct {
+	assistedServiceScheme string
+	assistedServiceHost   string
+	client                *http.Client
+}
+
+const fileRouteFormat = "/api/assisted-install/v2/infra-envs/%s/downloads/files"
+
+func NewAssistedServiceClient(assistedServiceScheme, assistedServiceHost, caCertFile string) *AssistedServiceClient {
+	client := &http.Client{}
+	if caCertFile != "" {
+		caCert, err := ioutil.ReadFile(caCertFile)
+		if err != nil {
+			log.Fatalf("Error opening cert file %s, %s", caCertFile, err)
```

Comment:
fatal will crash the service usually it recommended to return and error and handle it in main.

Reply:
This is just moved from https://github.com/openshift/assisted-image-service/pull/64/files#diff-5e672f28c4fa21931ec9a0d5438b1cf4deb15400b449463d9d8e77f879a6b0a2L53-L69

I think we should crash if we can't open this file, but if you'd like me to move the fatal log to main, I can do that.

Reply:
Not for this PR but lets open a ticket.

Reply:
:+1: opened https://issues.redhat.com/browse/MGMT-9538